### ISSUE TITLE
Hooks for post/pre compile

### ISSUE THREAD

**Issue by atticus-sullivan** [atticus-sullivan/cluttealtex#15]
Sometimes I like running in watch mode from my Makefile, but sometimes I need something more happening when recompiling (like the resulting document being postprocessed or copied to another location).

One solution would be to use something similar to [luamon](https://github.com/edubart/luamon), but the automatic detection of cluttealtex what files to watch for is pretty awesome and not something general tools like luamon can achieve.

So it might be worth to define some locations from which the user can run e.g. shell scripts.

Noteworthy locations:
- before first iteration
- directly before running pdflatex etc
- directly after running pdflatex etc
- after last iteration

We should also give some information about the build process (e.g. iteration-count, texengine, assembled command, outname/jobname, input file) via environment variables.

Another option would be to also run this with lua and pass all options somehow (serializing them to a lua-file which can be `dofile`ed for example).

**Comment by atticus-sullivan** [comment]
Will be (partially) implemented in #25. I'll close this issue after that PR is merged for now.

**Comment by atticus-sullivan** [comment]
With #28 in place, you can do
```lua
	cli_options = {
		echo_after_build = {
			long = "echo-after-build",
			handle_cli = function(options, _)
				io.stderr:write("hello world\n") -- printed when the argument is handled
				options.hooks.post_build[1] = {function(_)
					io.stderr:write("finished building =====================\n") -- only printed when the build succeeded
				end, "testing post_build hook"}
			end
		},
	},
```
in the config file. And then when executing as `cluttealtex --echo-after-build`, everything is in place.

This is exactly what I wanted. We'll see in the future, if the hook needs more arguments and also `pre_build` is not implemented in that PR (I personally also don't have a use for it tbh).
